Packages/ClientRuntime/Sources/Networking/Endpoint.swift

```diff
@@ -12,16 +12,49 @@ public struct Endpoint: Hashable {
     public let host: String
     public let port: Int16
     
+    public let headers: Headers?
+    public let properties: [String: AnyHashable]
+    
+    public init(urlString: String,
+                headers: Headers? = nil,
+                properties: [String: AnyHashable] = [:]) throws {
+        guard let url = URL(string: urlString) else {
+            throw ClientError.unknownError("invalid url \(urlString)")
+        }
+        
+        try self.init(url: url, headers: headers, properties: properties)
+    }
```

Comment:
**Correctness**: After some protracted discussion with the EP2.0 authors, it sounds like we should not allow users to set `properties` given that it has no defined keys/values or stable usage semantics. The `Endpoint` type still needs this field but it should only be settable by other code within the SDK (not from without).

Reply:
This is good intention vs what can be implemented. `Endpoint` struct is part of the `ClientRuntime` (Shared code for Generic SDK as well as AWS SDK). Without making it public interface, it is impossible to use across boundaries.

Reply:
The Kotlin SDK uses an annotation called `@InternalApi` to indicate when something is `public` by necessity but shouldn't be used by callers unless they're part of the SDK (in which case, they must opt-in at the callsite with another annotation).

Does Swift SDK have any similar capabilities? If not, how will users know that they should not set this field?

Reply:
There isn't a swift provided attribute. Also, saying it internal is incorrect because if a customer wants to implement their own EndpointResolver, they have to provide `properties` with exact same structure which we expect else it will not work.

We have discussed this many times to not use dictionary but that doesn't satisfy endpoints 2.0 requirements.